account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -219,75 +275,19 @@ export function alchemyGasAndPaymasterAndDataMiddleware(
       let erc20Context: RequestGasAndPaymasterAndDataRequest[0]["erc20Context"] =
         undefined;
       if (policyToken !== undefined) {
-        const maxAmountToken = policyToken.maxTokenAmount || maxUint256;
-
         erc20Context = {
           tokenAddress: policyToken.address,
           ...(policyToken.maxTokenAmount
             ? { maxTokenAmount: policyToken.maxTokenAmount }
             : {}),
         };
         if (policyToken.approvalMode === "PERMIT") {
-          // get a paymaster address
-          let paymasterAddress: Address | undefined = undefined;
-          const paymasterData = await (
-            client as AlchemySmartAccountClient
-          ).request({
-            method: "pm_getPaymasterStubData",
-            params: [
-              userOp,
-              account.getEntryPoint().address,
-              toHex(client.chain.id),
-              {
-                policyId: Array.isArray(policyId) ? policyId[0] : policyId,
-              },
-            ],
-          });
-
-          paymasterAddress = paymasterData.paymaster
-            ? paymasterData.paymaster
-            : paymasterData.paymasterAndData
-            ? sliceHex(paymasterData.paymasterAndData, 0, 20)
-            : undefined;
-
-          if (paymasterAddress === undefined || paymasterAddress === "0x") {
-            throw new Error("no paymaster contract address available");
-          }
-          const deadline = maxUint256;
-          const { data } = await client.call({
-            to: policyToken.address,
-            data: encodeFunctionData({
-              abi: parseAbi(EIP712NoncesAbi),
-              functionName: "nonces",
-              args: [account.address],
-            }),
-          });
-          if (!data) {
-            throw new Error("No nonces returned from erc20 contract call");
-          }
-
-          const typedPermitData = {
-            types: PermitTypes,
-            primaryType: "Permit" as const,
-            domain: {
-              name: policyToken.erc20Name ?? "",
-              version: policyToken.version ?? "",
-              chainId: BigInt(client.chain.id),
-              verifyingContract: policyToken.address,
-            } satisfies PermitDomain,
-            message: {
-              owner: account.address,
-              spender: paymasterAddress,
-              value: maxAmountToken,
-              nonce: BigInt(data),
-              deadline,
-            } satisfies PermitMessage,
-          } as const;
-
-          const signedPermit = await account.signTypedData(typedPermitData);
-          erc20Context.permit = encodeAbiParameters(
-            [{ type: "uint256" }, { type: "uint256" }, { type: "bytes" }],
-            [maxAmountToken, deadline, signedPermit]
+          erc20Context.permit = await generateSignedPermit(
+            userOp,
+            client as AlchemySmartAccountClient,
```

Comment:
actually... within the middleware this isn't an AlchemySmartAccountClient yet

Reply:
I see. amend this to MiddlewareClient